huma_signals/adapters/request_network/request_transaction_adapter.py

```diff
@@ -0,0 +1,109 @@
+from typing import ClassVar, List
+
+import pandas as pd
+import structlog
+import web3
+
+from huma_signals.adapters.ethereum_wallet import adapter as ethereum_wallet_adapter
+from huma_signals.adapters.polygon_wallet import adapter as polygon_wallet_adapter
+from huma_signals.adapters.request_network import models
+from huma_signals.commons import chains
+from huma_signals.domain.clients.request_client import request_client
+from huma_signals.settings import settings
+
+logger = structlog.get_logger(__name__)
+
+
+class RequestTransactionAdapter:
+    name: ClassVar[str] = "request_transaction"
+    required_inputs: ClassVar[List[str]] = [
+        "payer_address",
+        "payee_address",
+    ]
+    signals: ClassVar[List[str]] = list(
+        models.RequestTransactionSignals.__fields__.keys()
+    )
+
+    def __init__(
+        self,
+        request_client_: request_client.BaseRequestClient,
+        request_network_subgraph_endpoint_url: str = settings.request_network_subgraph_endpoint_url,
+        chain: chains.Chain = settings.chain,
+    ) -> None:
+        self.request_client = request_client_
+        self.request_network_subgraph_endpoint_url = (
+            request_network_subgraph_endpoint_url
+        )
+        self.chain = chain
+
+    async def fetch(
+        self,
+        payer_address: str,
+        payee_address: str,
+    ) -> models.RequestTransactionSignals:
+        if not web3.Web3.is_address(payer_address):
+            raise ValueError(f"Invalid payer address: {payer_address}")
+        if not web3.Web3.is_address(payee_address):
+            raise ValueError(f"Invalid payee address: {payee_address}")
+
+        payer_payments = await self.request_client.get_payments(
+            from_address=payer_address,
+            to_address=None,
+        )
+        payee_payments = await self.request_client.get_payments(
+            from_address=None,
+            to_address=payee_address,
+        )
+        payments_df = pd.DataFrame.from_records([*payer_payments, *payee_payments])
```

Comment:
There is a bit of duplicated code between the two request signal adapters. I don't think it's the right time to consolidate the two since the pre-approval one might evolve to something significantly different than the invoice based one due to its possible reliance on txn hashes instead of payer and payee addresses. It's a lot less costly to have duplication than to have the wrong abstraction.